### PR TITLE
irmin: fix fold for non-persisted, cleared trees

### PR DESCRIPTION
**Pull request by samoht**

### CODE REVIEW

**Review by CraigFe — APPROVED** [review]
LGTM

**Comment by Ngoguey42** [comment]
I noticed that some calls to `cached_hash` and `cached_value` were missing too:
https://github.com/mirage/irmin/commit/c53cb671cd96f45d4a80f4e87146682db8f7b7af

Do we want to apply that diff too?

**Comment by Ngoguey42** [comment]
I made 2 commits to fix the issues I raised above. I can push those here or open a second PR. 

https://github.com/Ngoguey42/irmin/commits/more-cached-calls

File: test/irmin/test_tree.ml [hunk 1]
```diff
@@ -398,24 +398,63 @@ let persist_tree : Store.tree -> Store.tree Lwt.t =
...
+      | `Node `Value -> Fmt.string ppf "value")
+    ( = )
+
+let pp_key = Irmin.Type.pp Store.Key.t
+
 let test_clear _ () =
-  let size = 830829 in
-  let* large_tree =
+  (* 1. Build a tree *)
+  let size = 8308 in
```
Comment (Ngoguey42) [inline test/irmin/test_tree.ml:414]: I shrinked the tree. Was it important?

Comment (samoht) [reply test/irmin/test_tree.ml:414]: yest it was to detect possible stack overflows :-)

**Pull request merged**